WIP adaptations for agreement 3.0

lepistone opened the pull request:
- [ ] remove temporary merge or OCA/purchase-workflow#113

lepistone committed:
fix ngo_purchase + agreement 3.0

lepistone committed:
framework_agreement_sourcing installable with fa 3

lepistone committed:
adapt tests to agreement-3

lepistone committed:
disable unported code to find a suitable FA

If found, the generated source is of type 'agreement'

lepistone committed:
disable automatic choice of agreement in source

lepistone committed:
adapt souced check to agreements 3.0

lepistone committed:
domain of a pricelist in view

lepistone committed:
adapt generation of FA PO

lepistone committed:
XXX pull OCA/purchase-workflow#113

lepistone committed:
do not set price when generating po from source

lepistone committed:
adapt sourcing check test

lepistone committed:
disable test: source w/ agreement if there is one

lepistone committed:
disable test: generate po from source

lepistone committed:
migrate some code to new API

lepistone committed:
fix creation of pricelist from requisition

See comment on how this should be improved, though

lepistone committed:
disable unported tests

We probably do not want to port those. A simple unit-style check on the
location should suffice.

lepistone committed:
bring in view changes from fa3

lepistone committed:
bring back two more test for agreement sourcing

lepistone committed:
fix po creation logic

Also, comment about disabled tests of an old wizard

lepistone committed:
in source, show price from agreement

lepistone committed:
use price from source in generated po

lepistone committed:
remove old broken ngo_view

lepistone committed:
remove dead code

This might have never been run: there is not action_confirm method in
the purchase order line (there is one in the line though). Anyway, it
refers to the framework_agreement_id field that is no more.

lepistone committed:
do not pass price_unit to onchange_product_id

Quite surprisingly, we always pass None as price_unit and name. If we
don't, the price stays at 0.0, and in the onchange_product_id, the price
is never computer.  All that comes from upstream:

https://github.com/odoo-dev/odoo/commit/7e454bef

lepistone committed:
pass order_type to onchange_product_id

Otherwise it falls back to the default 'rfq', and in that case we force
the price to zero (see purchase_rfq_bid_workflow).

lepistone committed:
pass state to onchange_partner_id, not 'draft'

Again, this tricks purchase_rfq_bid_workflow to think it's a rfq, so
prices are fixed at zero.

lepistone committed:
delete empty view

lepistone committed:
framework_agreement_tender readonly not in draft

lepistone committed:
bring in warning for portfolio without agreement

lepistone committed:
move pricelist field close to the supplier

This will bring up the portfolio, too. Makes sense since the portfolio
is related to the pricelist.

lepistone committed:
let users choose agreements manually

This is because of reflections on UI: users need to be able to choose
manually, even if agreement don't match exactly, and make their own
human evaluations.

yvaucher commented on framework_agreement_sourcing/model/logistic_requisition.py:
```diff
@@ -36,18 +32,21 @@ def _generate_default_source(self):
-            self.requisition_id.date,
-            self.requisition_id.incoterm_id.id,
-            self.requisition_id.incoterm_address,
+        req_date = datetime.strptime(self.requisition_id.date, DATETIME_FORMAT)
```
here you should use fields.Datetime.from_string()
https://www.odoo.com/documentation/8.0/reference/orm.html#openerp.fields.Datetime

yvaucher commented on framework_agreement_sourcing/tests/test_agreement_source_line_to_po.py:
```diff
@@ -35,29 +35,29 @@ def setUp(self):
+        self.wh = self.env.ref('stock.warehouse0')
 
-    def test_transform_source_to_agreement_wh_dest(self):
+    def XXX_unported_test_transform_source_to_agreement_wh_dest(self):
```
Isn't this WIP ?

lepistone committed:
fix attrs to integrate existing attrs

lepistone committed:
use new API for source fields

lepistone committed:
propose only relevant portfolios in sourcing

lepistone committed:
run the onchange to propagate agreement -> PO

This way, when fields like the incoterm are defined on the agreement,
they will take precedence on the version defined on the logistic
requisition.
